packages/ra-ui-materialui/src/list/filter/FilterForm.tsx

```diff
@@ -117,7 +117,15 @@ export const FilterFormBase = (props: FilterFormBaseProps) => {
     };
 
     const handleHide = useCallback(
-        event => hideFilter(event.currentTarget.dataset.key),
+        (event, onBeforeHideFilter) => {
+            if (
+                onBeforeHideFilter &&
+                'function' === typeof onBeforeHideFilter
+            ) {
+                return onBeforeHideFilter(event);
+            }
+            hideFilter(event.currentTarget.dataset.key);
+        },
```

Comment:
I'm not totally convinced by this API. Having the `onBeforeHideFilter` completely cancel the call to `hideFilter` is kind of surprising IMO. I would rather have suggested a pattern based on the return value, or the event propagation, to decide whether or not to call `hideFilter` after `onBeforeHideFilter`.

Besides, the issue is that this prop is documented nowhere (in the props sections of the inputs I mean), and is also not discoverable with TypeScript.